share/source.F90

```diff
@@ -42,128 +42,146 @@ function find_cell(cell_list, cell_list_size) result(icl_tmp)
 !             complex geometry. The results is suspicious under this condition.
 !           Skip it and resample next particle with error message.
 
+  use fixcom, only: mxa
+  use mcnp_global, only: ncl
+  use mcnp_params, only: dknd
+  use pblcom, only: xxx, yyy, zzz, erg
```

Comment:
The `xxx`, `yyy`, `zzz` and `erg` are also defined below. Are they different?

Reply:
Not sure what you mean. This line allows the `find_cell()` function to access those variables. The `source` subroutine doesn't automatically have access to the same variables as `find_cell()`, so they must be imported there too.

Reply:
Got it.